src/desdemona/ui/table.cljs

```diff
@@ -1,5 +1,39 @@
-(ns desdemona.ui.table)
+(ns desdemona.ui.table
+  (:require [wilson.dom :as wd]
+            [reagent.session :as session]))
+
+(def rows (:results @session/state))
+
+(def all-keys (-> (mapcat wd/get-all-keys rows) distinct wd/prepare-keys))
 
 (defn table-component
   []
-  [:div [:h2 "Table"]])
+  (let [sorted-rows (wd/sort-rows rows
+                                  {:default
+                                   (fn [k rows]
+                                     (if (= (:table-sort-order @session/state)
+                                            :asc)
+                                       (sort-by k rows)
+                                       (reverse (sort-by k rows))))}
+                                  (:table-sort-key @session/state))
+        get-new-order (fn [state k]
+                        (let [sort-key (:table-sort-key @session/state)
+                              sort-order (:table-sort-order @session/state)
+                              swap-order {:asc :desc :desc :asc}]
```

Comment:
If this takes the state, it should not deref `session/state`, that's why that API works the way it does. Instead of binding them in the let, consider destructuring state on line 19.